### PR TITLE
Dn api 175 public req attes met

### LINKED ISSUE #175: DOC - API Doc for Public methods on RequestForAttestation
https://github.com/KILTprotocol/sdk-js/pull/179

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## How to test:

Check docs:
Run npm run build:docs. This generates the doc at sdk-js/docs/api.
Open the so-generated files in your browser: sdk-js/docs/api/classes/requestforattestation.requestforattestation-1.html

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [x] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: src/requestforattestation/RequestForAttestation.ts [hunk 1]
```diff
@@ -164,6 +228,15 @@ export default class RequestForAttestation implements IRequestForAttestation {
     return this.verifySignature()
   }
 
+  /**
+   * Verifies the signature inside the [[RequestForAttestation]] object.
```
Comment (maudnals) [inline src/requestforattestation/RequestForAttestation.ts]: Could we briefly explain here what it means "business-wise" if the signature is wrong?

File: .eslintrc-jsdoc.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "semi": ["error", "always"],
     "prettier/prettier": ["warn", {
       "semi": true
-    }]
+    }],
+    "@typescript-eslint/no-unused-vars": "off"
```
Comment (maudnals) [inline .eslintrc-jsdoc.json]: Why is this one here?

File: src/requestforattestation/RequestForAttestation.ts [hunk 3]
```diff
@@ -60,6 +61,22 @@ function getHashRoot(leaves: Uint8Array[]): Uint8Array {
 }
 
 export default class RequestForAttestation implements IRequestForAttestation {
+  /**
+   * Builds an instance of [[RequestForAttestation]], from a simple object with the same properties.
```
Comment (maudnals) [inline src/requestforattestation/RequestForAttestation.ts]: Add [STATIC].

File: src/requestforattestation/RequestForAttestation.ts [hunk 4]
```diff
@@ -99,6 +116,29 @@ export default class RequestForAttestation implements IRequestForAttestation {
     this.claimerSignature = this.sign(identity)
   }
 
+  /**
+   * Removes [[Claim]] properties from the [[RequestForAttestation]] object.
```
Comment (maudnals) [inline src/requestforattestation/RequestForAttestation.ts]: Could we explain briefly when/why this is useful (privacy/...) - and maybe how it relates to `createPresentation`?

File: src/requestforattestation/RequestForAttestation.ts [hunk 5]
```diff
@@ -109,11 +149,35 @@ export default class RequestForAttestation implements IRequestForAttestation {
     })
   }
 
+  /**
+   * Removes the [[Claim]] Owner from the [[RequestForAttestation]] object.
```
Comment (maudnals) [inline src/requestforattestation/RequestForAttestation.ts]: Could we briefly explain when/why this is useful (privacy/...)?

File: src/requestforattestation/RequestForAttestation.ts (hunk 3 above)
Comment (Dudleyneedham) [reply src/requestforattestation/RequestForAttestation.ts]: Done.

File: src/requestforattestation/RequestForAttestation.ts (hunk 4 above)
Comment (Dudleyneedham) [reply src/requestforattestation/RequestForAttestation.ts]: Added the extra detail, makes it 100% better.

File: src/requestforattestation/RequestForAttestation.ts (hunk 5 above)
Comment (Dudleyneedham) [reply src/requestforattestation/RequestForAttestation.ts]: Again added more context to the sentence

File: src/requestforattestation/RequestForAttestation.ts [hunk 6]
```diff
@@ -60,6 +61,18 @@ function getHashRoot(leaves: Uint8Array[]): Uint8Array {
...
 export default class RequestForAttestation implements IRequestForAttestation {
+  /**
+   * [STATIC] Builds an instance of [[RequestForAttestation]], from a simple object with the same properties.
+   * Used for deserialization.
+   *
+   * @param obj - An object built from simple [[Claim]], [[Identity]] and legitimation objects.
+   * @returns  A new [[RequestForAttestation]] `object`.
+   * @example
+   * ```javascript
+   * // create an RequestForAttestation object, so we can call methods on it (`serialized` is a Claim, Identity and legitimation object )
```
Comment (Dudleyneedham) [inline src/requestforattestation/RequestForAttestation.ts]: Maybe this is too long. As I am thinking listing all the serialised object is too much

Comment (maudnals) [inline src/requestforattestation/RequestForAttestation.ts]: `serialized` is probably a serialized `RequestForAttestation` object

File: src/requestforattestation/RequestForAttestation.ts [hunk 7]
```diff
@@ -164,6 +207,16 @@ export default class RequestForAttestation implements IRequestForAttestation {
     return this.verifySignature()
   }
 
+  /**
+   * Verifies the signature inside the [[RequestForAttestation]] object, an unsuccessful signature will stop the Tx (Transaction).
+   *
+   * @returns Whether the verification of the claimers signature was successful.
+   * @example
+   * ```javascript
+   * RequestForAttestation.verifySignature();
```
Comment (maudnals) [inline src/requestforattestation/RequestForAttestation.ts]: Better, one thing though: `verifySignature` is not static, so it needs to be called on an instance aka a `requestForAttestation` (the capitalized version is reserved to designate the class).
Same for other instance methods: `verifyData`... etc.

**Pull request merged**